### PR TITLE
pdfstudio: fix hash mismatch

### LINKED ISSUE #360176: Build failure: pdfstudioviewer
## Steps To Reproduce

1. build pdfstudioviewer

## Build log

```
PDFStudioViewer_linux64.deb> trying https://download.qoppa.com/pdfstudioviewer/PDFStudioViewer_linux64.deb
PDFStudioViewer_linux64.deb>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
PDFStudioViewer_linux64.deb>                                  Dload  Upload   Total   Spent    Left  Speed
PDFStudioViewer_linux64.deb> 100  159M  100  159M    0     0   733k      0  0:03:43  0:03:43 --:--:--  668k
error: hash mismatch in fixed-output derivation '/nix/store/vq9zjqjqffmrh3kq2rfwryjcb023a5w7-PDFStudioViewer_linux64.deb.drv':
         specified: sha256-JQx5yJLjwW4VRXLM+/VNDXFN8ZcHJxlxyKDIzc++hEs=
            got:    sha256-hxReGuyGsBiEr7wWxWzQUQvxk11sgF9HkJ07L9i+e+0=
error: 1 dependencies of derivation '/nix/store/05mald3jk8zdw87k47bggdzb9hxp22gm-pdfstudioviewer-2023.0.3.drv' failed to build
error: 1 dependencies of derivation '/nix/store/1hwha0q8zbbv2gqqlvpgmin73hdm19ys-pdfstudioviewer-2023.0.3.drv' failed to build
```

## Additional context

Probably the package got updated, but this should be checked

## Metadata

```
 nix-info -m
 - system: `"x86_64-linux"`
 - host os: `Linux 6.11.10, NixOS, 25.05 (Warbler), 25.05.20241128.251e822`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.10`
 - channels(root): `"nixos-23.11"`
 - nixpkgs: `/nix/store/6adv0yxjghqny1ip70j1piib30fkqldg-source`
```

## Notify maintainers
@PhilippWoelfel 

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by ghost**
Fixes #360176
Close #287313

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by itepastra — CHANGES_REQUESTED** [review]
can you split the formatting commit and the actual update commits?

**Review by FliegendeWurst — COMMENTED** [review]
Why not use an archived download URL?

**Comment by Aleksanaa** [comment]
> There seems to be no stable download link for pdfstudioviewer

So you have to use an archived download url from e.g. webarchive.

File: pkgs/applications/misc/pdfstudio/default.nix [hunk 1]
```diff
@@ -8,90 +8,118 @@
...
+    inherit
+      desktopName
+      longDescription
+      pname
+      program
+      year
+      ;
+    version = "${year}.0.4";
     src = fetchurl {
       url = "https://download.qoppa.com/pdfstudio/v${year}/PDFStudio_v${dot2dash version}_linux64.deb";
```
Comment (Aleksanaa) [inline pkgs/applications/misc/pdfstudio/default.nix:96]: Also should use webarchive?

**Pull request merged**